### PR TITLE
all is done

### PR DESCRIPTION
**Pull request by ivlytvynets**

### CODE REVIEW

**Review by gigbat — APPROVED** [review]

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+language: java
+jdk:
+  - openjdk11
+script:
+  - mvn clean install
+cache:
+  directories:
+    - $HOME/.m2
```
Comment (Sofasmile) [inline .travis.yml]: empty line

File: pom.xml [hunk 2]
```diff
@@ -9,8 +9,75 @@
...
+                        <source>${jdk.version}</source>
+                        <target>${jdk.version}</target>
+                        <encoding>${project.build.sourceEncoding}</encoding>
+                    </configuration>
+                </plugin>
+            </plugins>
+        </pluginManagement>
+    </build>
+
 </project>
```
Comment (Sofasmile) [inline pom.xml]: empty line

File: src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+
+@Dao
+public class MovieDaoImpl implements MovieDao {
+    @Override
+    public Movie add(Movie movie) {
+        Transaction transaction = null;
+        Session session = HibernateUtil.getSessionFactory().openSession();
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java]: we should open session in the try block

File: src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        try {
+            transaction = session.beginTransaction();
+            Long movieId = (Long) session.save(movie);
+            transaction.commit();
+            movie.setId(movieId);
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java]: https://mate-academy.github.io/jv-program-fulltime/03_hibernate/checklist/hibernate_01.html

File: src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't insert Movie entity", e);
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java]:
```suggestion
            throw new DataProcessingException("Can't insert Movie entity " + movie, e);
```

File: src/main/java/com/dev/cinema/util/HibernateUtil.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    private static SessionFactory sessionFactory = initSessionFactory();
+
+    private HibernateUtil(){
+    }
+
+    private static SessionFactory initSessionFactory() {
+        try {
+            return new Configuration().configure().buildSessionFactory();
+        } catch (Exception e) {
+            throw new SessionException("Error creating Session Factory", e);
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/util/HibernateUtil.java]:
```suggestion
            throw new SessionFactoryInitializationException("Error creating Session Factory", e);
```

File: src/main/resources/hibernate.cfg.xml [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+        <property name="connection.url">jdbc:mysql://localhost:3306/cinema</property>
+        <property name="connection.driver_class">com.mysql.cj.jdbc.Driver</property>
+        <property name="connection.username">ivanlytvynets</property>
+        <property name="connection.password">Vanya_2002</property>
+        <property name="show_sql">true</property>
+        <property name="hibernate.hbm2ddl.auto">create-drop</property>
+
+        <mapping class="com.dev.cinema.model.Movie"/>
+    </session-factory>
+</hibernate-configuration>
```
Comment (Sofasmile) [inline src/main/resources/hibernate.cfg.xml]: empty line

**Review by Sofasmile — CHANGES_REQUESTED** [review]

File: pom.xml [hunk 8]
```diff
@@ -9,8 +9,74 @@
...
+        <dependency>
+            <groupId>mysql</groupId>
+            <artifactId>mysql-connector-java</artifactId>
+            <version>8.0.22</version>
+        </dependency>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>4.12</version>
+            <scope>test</scope>
```
Comment (RomanDubovskyi) [inline pom.xml]: If we don't have tests let's not add this dependency

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**